### PR TITLE
Replace local file storage with S3 file storage

### PR DESCRIPTION
**Pull request by hakbailey**
Also updates tests to match Python version and use pipenv with tox

### CODE REVIEW

File: ebooks/queries.py [hunk 1]
```diff
@@ -1,26 +1,38 @@
...
-    files = os.listdir(files_path)
+    for key in bucket.objects.all():
+        files.append(key.key)
 
     for f in files:
         try:
             p = f.index('.')
             item_name = f[:p]
-        except:
+        except Exception:
```
Comment (gravesm) [inline ebooks/queries.py]: You should explicitly check for a `ValueError` here, and down on line 35.

File: ebooks/queries.py [hunk 2]
```diff
@@ -1,26 +1,38 @@
...
+    session = Session(aws_access_key_id=AWS_ACCESS_KEY_ID,
+                      aws_secret_access_key=AWS_SECRET_ACCESS_KEY,
+                      region_name=AWS_REGION_NAME)
+    s3 = session.resource('s3')
+    bucket = s3.Bucket(AWS_BUCKET_NAME)
 
-    base_path = os.path.dirname(os.path.abspath(__file__))
-    files_path = os.path.join(base_path, 'static/files')
-    files = os.listdir(files_path)
+    for key in bucket.objects.all():
```
Comment (gravesm) [inline ebooks/queries.py]: If there aren't many objects in the bucket, I think it's fine to do things as you have. In general, I'd avoid looping through a list twice, though. A nice, pythonic solution here would be to use the built-in `filter()` function.

File: ebooks/queries.py [hunk 3]
```diff
@@ -1,26 +1,38 @@
...
+AWS_ACCESS_KEY_ID = os.getenv('AWS_ACCESS_KEY_ID')
+AWS_SECRET_ACCESS_KEY = os.getenv('AWS_SECRET_ACCESS_KEY')
+AWS_REGION_NAME = os.getenv('AWS_REGION_NAME')
+AWS_BUCKET_NAME = os.getenv('AWS_BUCKET_NAME')
+
 
 def get_filenames(file_id):
     RESULTS = []
     files = []
+    session = Session(aws_access_key_id=AWS_ACCESS_KEY_ID,
```
[the remark this answers is not in the document]
Comment (hakbailey) [reply ebooks/queries.py]: Fixed in latest commit

File: ebooks/queries.py (hunk 1 above)
Comment (hakbailey) [reply ebooks/queries.py]: No longer needed

File: ebooks/queries.py (hunk 2 above)
Comment (hakbailey) [reply ebooks/queries.py]: Updated, thanks!

**Pull request merged**